### PR TITLE
Attach interface/device with virtio model type

### PR DESCRIPTION
**Pull request by yalzhang**
Modify the attach options with model type=virtio

Virtio model type is recommend, so add it as the attach attribute.

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 1]
```diff
@@ -266,7 +266,8 @@ def run(test, params, env):
             if vm.is_alive():
                 vm.destroy()
         else:
-            vm.wait_for_login().close()
+            session = vm.wait_for_login()
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: no need change,just keep previous one

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 2]
```diff
@@ -283,7 +284,8 @@ def run(test, params, env):
 
         # Set attach-interface options and Start attach-interface test
         if correct_attach:
-            options = set_options("network", "default", iface_mac, "", "attach")
+            options = set_options("network", "default", iface_mac, "",
+                                  "attach", None, iface_model)
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: I assume None should be '--model' option

**Comment by yalzhang** [comment]
Run all the cases, FAIL 4 is because an existing bug, but I will not add the bug info into the fail message because there maybe other reason to cause the same error. Just leave it as it is.
The CANCEL 7 part is expected on new version of rhel, I will deal it in another pr.

$ avocado run --vt-type libvirt  attach_detach_interface attach_detach_interface_matrix
JOB ID     : 425b8f1be1cc75deb4f6b7f8a647dd294af49870
JOB LOG    : /root/avocado/job-results/job-2019-01-04T09.55-425b8f1/job.log
 (001/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.default_network.domname: PASS (39.77 s)
 (002/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.default_network.domid: FAIL: ['Attach Failed: error: Failed to attach interface\nerror: internal error: No more available PCI slots\n'] (23.80 s)
 (003/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.default_network.domuuid: PASS (40.64 s)
 (004/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.exist_bridge.domname: CANCEL: Command 'brctl' is missing. You must install it. (23.55 s)
 (005/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.exist_bridge.domid: CANCEL: Command 'brctl' is missing. You must install it. (23.52 s)
 (006/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.exist_bridge.domuuid: CANCEL: Command 'brctl' is missing. You must install it. (23.52 s)
 (007/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.default.domname: PASS (45.85 s)
 (008/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.default.domid: FAIL: ['Attach Failed: error: Failed to attach interface\nerror: internal error: No more available PCI slots\n'] (26.58 s)
 (009/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.default.domuuid: PASS (45.54 s)
 (010/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.print_xml.domname: PASS (26.47 s)
 (011/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.print_xml.domid: PASS (26.67 s)
 (012/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.print_xml.domuuid: PASS (26.59 s)
 (013/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_running.direct.at_detach_device.domname: PASS (46.59 s)
 (014/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.default_network.domname: PASS (40.67 s)
 (015/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.default_network.domid: FAIL: ['Attach Failed: error: Failed to attach interface\nerror: internal error: No more available PCI slots\n'] (23.78 s)
 (016/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.default_network.domuuid: PASS (40.69 s)
 (017/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.exist_bridge.domname: CANCEL: Command 'brctl' is missing. You must install it. (23.56 s)
 (018/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.exist_bridge.domid: CANCEL: Command 'brctl' is missing. You must install it. (23.49 s)
 (019/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.exist_bridge.domuuid: CANCEL: Command 'brctl' is missing. You must install it. (23.52 s)
 (020/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.direct.default.domname: PASS (45.35 s)
 (021/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.direct.default.domid: FAIL: ['Attach Failed: error: Failed to attach interface\nerror: internal error: No more available PCI slots\n'] (26.55 s)
 (022/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.direct.default.domuuid: PASS (45.55 s)
 (023/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.direct.print_xml.domname: PASS (26.55 s)
 (024/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.direct.print_xml.domid: PASS (26.58 s)
 (025/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.normal_test.vm_paused.direct.print_xml.domuuid: PASS (26.61 s)
 (026/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_unknow.none: PASS (23.51 s)
 (027/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_unknow.hex_domid: PASS (23.50 s)
 (028/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_unknow.invalid_uuid: PASS (23.50 s)
 (029/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_unknow.domname: PASS (23.51 s)
 (030/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_none.none: PASS (23.54 s)
 (031/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_none.hex_domid: PASS (23.50 s)
 (032/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_none.invalid_uuid: PASS (23.54 s)
 (033/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_unkown.type_none.domname: PASS (23.55 s)
 (034/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_unknow.none: PASS (23.52 s)
 (035/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_unknow.hex_domid: PASS (23.51 s)
 (036/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_unknow.invalid_uuid: PASS (23.54 s)
 (037/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_unknow.domname: PASS (23.55 s)
 (038/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_none.none: PASS (23.55 s)
 (039/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_none.hex_domid: PASS (23.47 s)
 (040/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_none.invalid_uuid: PASS (23.55 s)
 (041/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_iface.source_none.type_none.domname: PASS (23.55 s)
 (042/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_unknow.none: PASS (30.11 s)
 (043/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_unknow.hex_domid: PASS (30.18 s)
 (044/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_unknow.invalid_uuid: PASS (30.20 s)
 (045/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_unknow.domname: PASS (30.19 s)
 (046/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_none.none: PASS (30.22 s)
 (047/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_none.hex_domid: PASS (30.17 s)
 (048/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_none.invalid_uuid: PASS (30.13 s)
 (049/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_none.domname: PASS (30.18 s)
 (050/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.detach_iface.type_none.readonly: PASS (23.53 s)
 (051/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_with_script.type_eth.non_exist: PASS (23.54 s)
 (052/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.attach_with_script.type_br.non_exist: CANCEL: Command 'brctl' is missing. You must install it. (23.34 s)
 (053/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface.error_test.vm_shutoff: PASS (28.79 s)
 (054/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live.dt_option_live.at_okay_dt_okay: PASS (50.51 s)
 (055/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live.dt_option_config.at_okay_dt_error: PASS (44.57 s)
 (056/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live.dt_option_current.at_okay_dt_okay: PASS (50.62 s)
 (057/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live.dt_option_default.at_okay_dt_okay: PASS (50.65 s)
 (058/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live.dt_option_live_config.at_okay_dt_error: PASS (45.06 s)
 (059/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_config.dt_option_live.at_okay_dt_error: PASS (44.96 s)
 (060/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_config.dt_option_config.at_okay_dt_okay: PASS (45.09 s)
 (061/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_config.dt_option_current.at_okay_dt_error: PASS (45.00 s)
 (062/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_config.dt_option_default.at_okay_dt_error: PASS (45.01 s)
 (063/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_config.dt_option_live_config.at_okay_dt_error: PASS (44.49 s)
 (064/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_current.dt_option_live.at_okay_dt_okay: PASS (50.19 s)
 (065/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_current.dt_option_config.at_okay_dt_error: PASS (45.06 s)
 (066/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_current.dt_option_current.at_okay_dt_okay: PASS (50.59 s)
 (067/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_current.dt_option_default.at_okay_dt_okay: PASS (50.16 s)
 (068/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_current.dt_option_live_config.at_okay_dt_error: PASS (44.54 s)
 (069/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_default.dt_option_live.at_okay_dt_okay: PASS (50.09 s)
 (070/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_default.dt_option_config.at_okay_dt_error: PASS (44.57 s)
 (071/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_default.dt_option_current.at_okay_dt_okay: PASS (50.66 s)
 (072/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_default.dt_option_default.at_okay_dt_okay: PASS (50.08 s)
 (073/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_default.dt_option_live_config.at_okay_dt_error: PASS (44.60 s)
 (074/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live_config.dt_option_live.at_okay_dt_okay: PASS (50.68 s)
 (075/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live_config.dt_option_config.at_okay_dt_okay: PASS (65.79 s)
 (076/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live_config.dt_option_current.at_okay_dt_okay: PASS (50.38 s)
 (077/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live_config.dt_option_default.at_okay_dt_okay: PASS (50.62 s)
 (078/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_running.at_option_live_config.dt_option_live_config.at_okay_dt_okay: PASS (50.70 s)
 (079/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live.dt_option_live.at_error_dt_error: PASS (27.01 s)
 (080/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live.dt_option_config.at_error_dt_error: PASS (26.94 s)
 (081/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live.dt_option_current.at_error_dt_error: PASS (26.96 s)
 (082/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live.dt_option_default.at_error_dt_error: PASS (26.94 s)
 (083/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live.dt_option_live_config.at_error_dt_error: PASS (26.88 s)
 (084/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_config.dt_option_live.at_okay_dt_error: PASS (27.30 s)
 (085/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_config.dt_option_config.at_okay_dt_okay: PASS (27.85 s)
 (086/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_config.dt_option_current.at_okay_dt_okay: PASS (27.85 s)
 (087/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_config.dt_option_default.at_okay_dt_error: PASS (27.25 s)
 (088/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_config.dt_option_live_config.at_okay_dt_error: PASS (27.32 s)
 (089/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_current.dt_option_live.at_okay_dt_error: PASS (27.31 s)
 (090/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_current.dt_option_config.at_okay_dt_okay: PASS (27.81 s)
 (091/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_current.dt_option_current.at_okay_dt_okay: PASS (27.88 s)
 (092/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_current.dt_option_default.at_okay_dt_error: PASS (27.31 s)
 (093/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_current.dt_option_live_config.at_okay_dt_error: PASS (27.28 s)
 (094/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_default.dt_option_live.at_error_dt_error: PASS (27.00 s)
 (095/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_default.dt_option_config.at_error_dt_error: PASS (26.91 s)
 (096/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_default.dt_option_current.at_error_dt_error: PASS (26.98 s)
 (097/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_default.dt_option_default.at_error_dt_error: PASS (26.96 s)
 (098/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_default.dt_option_live_config.at_error_dt_error: PASS (27.00 s)
 (099/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live_config.dt_option_live.at_error_dt_error: PASS (26.95 s)
 (100/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live_config.dt_option_config.at_error_dt_error: PASS (27.01 s)
 (101/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live_config.dt_option_current.at_error_dt_error: PASS (26.94 s)
 (102/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live_config.dt_option_default.at_error_dt_error: PASS (26.95 s)
 (103/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_shutoff.at_option_live_config.dt_option_live_config.at_error_dt_error: PASS (26.90 s)
 (104/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live.dt_option_live.at_okay_dt_okay: PASS (53.30 s)
 (105/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live.dt_option_config.at_okay_dt_error: PASS (47.67 s)
 (106/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live.dt_option_current.at_okay_dt_okay: PASS (53.13 s)
 (107/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live.dt_option_default.at_okay_dt_okay: PASS (53.27 s)
 (108/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live.dt_option_live_config.at_okay_dt_error: PASS (47.69 s)
 (109/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_config.dt_option_live.at_okay_dt_error: PASS (47.50 s)
 (110/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_config.dt_option_config.at_okay_dt_okay: PASS (48.15 s)
 (111/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_config.dt_option_current.at_okay_dt_error: PASS (47.58 s)
 (112/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_config.dt_option_default.at_okay_dt_error: PASS (47.56 s)
 (113/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_config.dt_option_live_config.at_okay_dt_error: PASS (47.60 s)
 (114/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_current.dt_option_live.at_okay_dt_okay: PASS (53.18 s)
 (115/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_current.dt_option_config.at_okay_dt_error: PASS (47.64 s)
 (116/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_current.dt_option_current.at_okay_dt_okay: PASS (53.22 s)
 (117/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_current.dt_option_default.at_okay_dt_okay: PASS (53.12 s)
 (118/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_current.dt_option_live_config.at_okay_dt_error: PASS (47.69 s)
 (119/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_default.dt_option_live.at_okay_dt_okay: PASS (53.19 s)
 (120/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_default.dt_option_config.at_okay_dt_error: PASS (47.60 s)
 (121/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_default.dt_option_current.at_okay_dt_okay: PASS (53.23 s)
 (122/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_default.dt_option_default.at_okay_dt_okay: PASS (53.14 s)
 (123/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_default.dt_option_live_config.at_okay_dt_error: PASS (47.74 s)
 (124/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live_config.dt_option_live.at_okay_dt_okay: PASS (53.11 s)
 (125/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live_config.dt_option_config.at_okay_dt_okay: PASS (68.47 s)
 (126/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live_config.dt_option_current.at_okay_dt_okay: PASS (53.37 s)
 (127/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live_config.dt_option_default.at_okay_dt_okay: PASS (53.21 s)
 (128/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_paused.at_option_live_config.dt_option_live_config.at_okay_dt_okay: PASS (53.26 s)
 (129/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live.dt_option_live.at_okay_dt_okay: PASS (61.93 s)
 (130/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live.dt_option_config.at_okay_dt_error: PASS (56.64 s)
 (131/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live.dt_option_current.at_okay_dt_okay: PASS (62.08 s)
 (132/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live.dt_option_default.at_okay_dt_okay: PASS (61.96 s)
 (133/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live.dt_option_live_config.at_okay_dt_error: PASS (56.62 s)
 (134/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_config.dt_option_live.at_error_dt_error: PASS (46.08 s)
 (135/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_config.dt_option_config.at_error_dt_error: PASS (46.03 s)
 (136/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_config.dt_option_current.at_error_dt_error: PASS (46.03 s)
 (137/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_config.dt_option_default.at_error_dt_error: PASS (46.09 s)
 (138/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_config.dt_option_live_config.at_error_dt_error: PASS (46.04 s)
 (139/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_current.dt_option_live.at_okay_dt_okay: PASS (62.10 s)
 (140/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_current.dt_option_config.at_okay_dt_error: PASS (56.67 s)
 (141/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_current.dt_option_current.at_okay_dt_okay: PASS (62.07 s)
 (142/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_current.dt_option_default.at_okay_dt_okay: PASS (62.00 s)
 (143/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_current.dt_option_live_config.at_okay_dt_error: PASS (56.56 s)
 (144/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_default.dt_option_live.at_okay_dt_okay: PASS (62.10 s)
 (145/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_default.dt_option_config.at_okay_dt_error: PASS (56.66 s)
 (146/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_default.dt_option_current.at_okay_dt_okay: PASS (62.08 s)
 (147/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_default.dt_option_default.at_okay_dt_okay: PASS (61.91 s)
 (148/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_default.dt_option_live_config.at_okay_dt_error: PASS (56.64 s)
 (149/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live_config.dt_option_live.at_error_dt_error: PASS (46.07 s)
 (150/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live_config.dt_option_config.at_error_dt_error: PASS (46.04 s)
 (151/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live_config.dt_option_current.at_error_dt_error: PASS (46.07 s)
 (152/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live_config.dt_option_default.at_error_dt_error: PASS (46.02 s)
 (153/153) type_specific.io-github-autotest-libvirt.virsh.attach_detach_interface_matrix.pre_vm_state_transient.at_option_live_config.dt_option_live_config.at_error_dt_error: PASS (46.10 s)
RESULTS    : PASS 142 | ERROR 0 | FAIL 4 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 7
JOB TIME   : 6053.37 s

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py [hunk 3]
```diff
@@ -24,6 +24,8 @@ def set_options(iface_type=None, iface_source=None,
         options += " --source '%s'" % iface_source
     if iface_mac is not None:
         options += " --mac '%s'" % iface_mac
+    if iface_model is not None and operation_type == "attach":
```
Comment (chloerh) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py]: if the var 'iface_model' could not be set to empty str, list or False or 0
if iface_model and operation_type == "attach": 
should be enough

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 4]
```diff
@@ -89,7 +89,7 @@ def login_to_check(vm, checked_mac):
         session = vm.wait_for_login()
     except Exception as detail:  # Do not care Exception's type
         return (1, "Can not login to vm:%s" % detail)
-    status, output = session.cmd_status_output("ip -4 -o link list")
+    status, output = session.cmd_status_output("sleep 5; ip -4 -o link list")
```
[the remark this answers is not in the document]
Comment (yalzhang) [reply libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: The shell command is planned to run on guest to confirm the guest can get dhcp ip address after boot. It is ok to use time.sleep(5) on host instead before the ip check command. Updated

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py (hunk 3 above)
Comment (yalzhang) [reply libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface_matrix.py]: Updated and Thank you!

**Comment by chloerh** [comment]
Looks good to me

File: libvirt/tests/cfg/virsh_cmd/domain/virsh_attach_detach_interface.cfg [hunk 5]
```diff
@@ -11,10 +11,17 @@
             # Domain types for test
                 - domname:
                     at_detach_iface_vm_ref = "domname"
+                    at_detach_iface_model = "virtio"
                 - domid:
                     at_detach_iface_vm_ref = "domid"
+                    at_detach_iface_model = "rtl8139"
+                    machine_type = pseries:
```
Comment (chloerh) [inline libvirt/tests/cfg/virsh_cmd/domain/virsh_attach_detach_interface.cfg]: Another '=' is missing, also in line:23

File: libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py [hunk 6]
```diff
@@ -89,6 +89,7 @@ def login_to_check(vm, checked_mac):
         session = vm.wait_for_login()
     except Exception as detail:  # Do not care Exception's type
         return (1, "Can not login to vm:%s" % detail)
+    time.sleep(5)
```
Comment (kylazhang) [inline libvirt/tests/src/virsh_cmd/domain/virsh_attach_detach_interface.py]: Can we use wait_for?

**Pull request merged**